core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -516,38 +541,42 @@ sealed trait ZPure[-S1, +S2, -R, +E, +A] { self =>
   final def run(s: S1)(implicit ev1: Any <:< R, ev2: E <:< Nothing): (S2, A) =
```

Comment:
With the additional functionality we are adding there is a risk of having a ton of different ways to `run` a `ZPure` value. Also, I would like to improve the ergonomics in the many cases where there is no state type.

I was thinking about trying to make the following the primary method to run `ZPure` values:

```scala
final def run(implicit ev1: Unit <:< S1, ev2: Any <:< R, ev3: E <:< Nothing): A
```

Then we could have a `runLog` variant that exposes the value and the log. Users would provide the input state through a `provideState` operator if they had not already set an initial state, similar to how you provide the environment to a `ZIO` effect before running it, and would normally handle their errors before running it, even by just doing `zPure.either`. We could perhaps have a `runEither` variant and maybe a `runState` variant for when you are just doing state but I would like to avoid users having to always provide a `Unit` state type when running computations and having every possible variant of state, logging, error, and value outputs.